bpf/lib/nat.h

```diff
@@ -1442,13 +1743,20 @@ snat_v6_rev_nat(struct __ctx_buff *ctx, const struct ipv6_nat_target *target)
 		return NAT_PUNT_TO_STACK;
 	};
 
-	if (snat_v6_rev_nat_can_skip(target, &tuple))
-		return NAT_PUNT_TO_STACK;
-	ret = snat_v6_rev_nat_handle_mapping(ctx, &tuple, &state, off, target);
-	if (ret > 0)
-		return CTX_ACT_OK;
-	if (ret < 0)
-		return ret;
+	if (tuple.nexthdr != nexthdr) {
+		/* Context of inner packet rev-NATed. Switch back to
+		 * the outer header.
+		 */
+		tuple.nexthdr = nexthdr;
+	} else {
```

Comment:
Just do this unconditionally inside the `ICMPV6_PKT_TOOBIG` case, and use a `goto`  to skip right to `snat_v6_rewrite_ingress()` ?

Reply:
++